utility/utils.js

```diff
@@ -130,14 +130,13 @@ class Utils {
    * @inner */
   static async ensureAuthenticated(req, res, next) {
     if (
-      req.body[0] != null &&
-      req.body[0].user_api != null &&
-      req.body[0].user_id
+      req.get("user-id") &&
+      req.get("user-api")
     ) {
       // Check the user based on API.
-      let apiKey = req.body[0].user_api;
-      let userId = req.body[0].user_id;
-      let sqlQuery = "SELECT * FROM user WHERE id = ?";
+      const apiKey = req.get("user-api");
```

Comment:
Maybe something like
`const apiKey = req.get("user-api") == null ? req.body[0].user_api : req.get("user-api")` That way it should satisfy both conditions.

Reply:
See top conversation